tests/result_test.py

```diff
@@ -132,3 +133,73 @@ def test_detector_results_are_read_only():
         res.detectors['detector2'] = detector
     with pytest.raises(TypeError, match='object does not support item deletion'):
         del res.detectors['detector']
+
+
+def test_component_results_data_access():
+    topen = 10.0 * ms
+    tclose = 20.0 * ms
+    chopper = make_chopper(
+        topen=[topen],
+        tclose=[tclose],
+        f=10.0 * Hz,
+        phase=0.0 * deg,
+        distance=10 * meter,
+        name='chopper',
+    )
+    detector = tof.Detector(distance=20 * meter, name='detector')
+
+    source = make_source(
+        arrival_times=sc.concat(
+            [0.9 * topen, 0.5 * (topen + tclose), 1.1 * tclose], dim='event'
+        ),
+        distance=chopper.distance,
+        pulses=3,
+        frequency=14.0 * Hz,
+    )
+
+    model = tof.Model(source=source, choppers=[chopper], detectors=[detector])
+    res = model.run()
+    ch = res.choppers['chopper']
+    det = res.detectors['detector']
+
+    for field in ('tofs', 'wavelengths', 'birth_times', 'speeds'):
+        assert 'visible' in getattr(ch, field).data
+        assert 'blocked' in getattr(ch, field).data
+        assert 'visible' in getattr(det, field).data
+        assert 'blocked' not in getattr(det, field).data
+
+    assert list(ch.tofs.visible.data.keys()) == [f'pulse:{i}' for i in range(3)]
+    assert list(det.wavelengths.visible.data.keys()) == [f'pulse:{i}' for i in range(3)]
+
+
+def test_component_results_data_slicing():
+    topen = 10.0 * ms
+    tclose = 20.0 * ms
+    chopper = make_chopper(
+        topen=[topen],
+        tclose=[tclose],
+        f=10.0 * Hz,
+        phase=0.0 * deg,
+        distance=10 * meter,
+        name='chopper',
+    )
+    detector = tof.Detector(distance=20 * meter, name='detector')
+
+    source = make_source(
```

Comment:
Have you considered defining, e.g., a Pytest fixture for the `source`? It seems to be repeated.